city_scrapers/spiders/il_governors_state_university.py

```diff
@@ -0,0 +1,201 @@
+
+    def _normalize_date(self, date, default_year):
+        """The dates appear in pretty variable formats, including in some cases without a year.
+        This method normalizes."""
+        clean_date = date.replace(",", "").replace(".", "").lower().strip()
+        # There was a stray "sept." in the data, although usually the month is
+        # fully spelled out. Use first three chars of the date string to get the month.
+        months = [
+            "january",
+            "february",
+            "march",
+            "april",
+            "may",
+            "june",
+            "july",
+            "august",
+            "september",
+            "october",
+            "november",
+            "december",
+        ]
+        month_map = {m[:3]: m for m in months}
+        month, day, year = re.findall(
+            r"([a-z]+)\.?\s+(\d\d?),?\s*(\d\d\d\d)?", clean_date
+        )[0]
+        month = month_map[month[:3]]
+        year = year if len(year) == 4 else default_year
+        return f"{month} {day} {year}"
+
+    def _normalize_time(self, time_str):
+        """Normalize time format. Sometimes it comes with colons or periods,
+        sometimes not"""
+        times = re.findall(self.time_re, time_str)
+        if len(times) == 0:
+            return None
+        hour, minute, ampm = times[0]
+        if len(minute.strip(":")) > 0:
+            minute = minute.strip(":")
+        else:
+            minute = "00"
+        ampm = ampm.replace(".", "")
+        return f"{hour}:{minute} {ampm}"
+
+    def _parse_start(self, item, default_year):
+        """Parse start datetime as a naive datetime object."""
+
+        # try to find the date in the first column, and if it isn't there, fall back
+        # to the third
+        day = " ".join(item[0].css("* ::text").getall())
+        if not re.search(r"\d", day):
+            day = " ".join(item[2].css("* ::text").getall())
+        clean_day = self._normalize_date(day, default_year)
+        time = " ".join(item[1].css("* ::text").getall()).lower()
+        clean_time = self._normalize_time(time)
+        if clean_time is not None:
+            return datetime.strptime(f"{clean_day} {clean_time}", "%B %d %Y %I:%M %p")
+        # fall back to midnight if no time specified
+        return datetime.strptime(clean_day, "%B %d %Y")
+
+    def _parse_end(self, item):
+        """Parse end datetime as a naive datetime object. Added by pipeline if None.
+        Not available for this website."""
+        return None
+
+    def _parse_time_notes(self, item):
+        """Parse any additional notes on the timing of the meeting"""
+        return ""
+
+    def _parse_all_day(self, item):
+        """Parse or generate all-day status. Defaults to False. Doesn't seem to occur
+        for this website, with the possible exception of the retreats, which aren't
+        quite all day"""
+        return False
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
```

Comment:
It looks like this is picking up "Postponed to" as the location name, so it might be easiest to return the default if "Postponed" is present. It shows up in [this line in the output](https://github.com/City-Bureau/city-scrapers/runs/1180091014#step:10:372). Let me know if that link doesn't work, I haven't done many direct links to GitHub Actions output

I'm also seeing one where "Room D34000" shows up as the address [in this output line](https://github.com/City-Bureau/city-scrapers/runs/1180091014#step:10:832) so we might be able to use the default address if "Room" is in the address here.